### ISSUE TITLE
Google Colab CUDA Issue

### ISSUE THREAD

**Issue by DavidLandup0** [DavidLandup0/deepvision#84]
When porting weights, Google Colab's T4 runs out of memory, due to parameter porting?
Doesn't happen locally on a 6GB GPU (1660Super) but does on a 16GB GPU (T4).

**Comment by DavidLandup0** [comment]
Turns out this happens when TF and PT share a session. TF hogs the available memory and PT can't use it.
Simple fix is to not run weight porting and PyTorch fine-tuning in the same session. I.e. save the new PT `state_dict()` and reload it in another session.

Why does it work locally on a 1660Super though?

**Comment by DavidLandup0** [comment]
Further debugging points to these lines being the issue:

```
for pt_block, tf_block in zip(target_model.blocks, tf_blocks):
    if isinstance(tf_block, __FusedMBConvTF):
        converted_block = fused_mbconv.tf_to_pt(tf_block)
        pt_block.load_state_dict(converted_block.state_dict())
    if isinstance(tf_block, __MBConvTF):
        converted_block = mbconv.tf_to_pt(tf_block)
        pt_block.load_state_dict(converted_block.state_dict())
```

**Comment by DavidLandup0** [comment]
The issue was calling `layer(dummy_input)` in each respective `tf_to_pt()` function, without calling top-level `model(dummy_input)`. Not exactly sure why this was an issue, but it clearly was.
Fixed now with #85
